pkg/params/params.go

```diff
@@ -0,0 +1,354 @@
+func (p *Params) Get(key string) *Param {
+	for _, param := range *p {
+		if key == param.Key {
+			return param
+		}
+	}
+	return nil
+}
+
+func (p *Params) Set(key, val string) error {
+	for _, e := range *p {
+		if e.Key == key {
+			return e.Set(val)
+		}
+	}
+	return fmt.Errorf("param %q not found", key)
+}
+
+func (p *Params) ParamMap() (res map[string]string) {
+	res = make(map[string]string)
+	for _, v := range *p {
+		res[v.Key] = v.String()
+	}
+	return
+}
+
+func (p *Params) ValidateStringMap(cfg map[string]string) error {
+	for _, param := range *p {
+		value, ok := cfg[param.Key]
+		if !ok && param.IsMandatory {
+			return fmt.Errorf("expected value for %q", param.Key)
+		}
+		if param.Validator != nil {
+			if err := param.Validator(value); err != nil {
+				return fmt.Errorf("invalid value %q as %q: %w", value, param.Key, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (p Collection) Set(entry, key, val string) error {
+	if _, ok := p[entry]; !ok {
+		return fmt.Errorf("%q is not part of the collection", entry)
+	}
+	return p[entry].Set(key, val)
+}
+
+// String is a member of the pflag.Value interface, which is used by cobra
+func (p *Param) String() string {
+	if p == nil {
+		return ""
+	}
+	if p.assigned {
+		return p.value
+	}
+	return p.DefaultValue
+}
+
+// Set validates and sets the new value; it is also a member of the pflag.Value interface,
+// which is used by cobra
+func (p *Param) Set(val string) error {
+	err := p.Validate(val)
+	if err != nil {
+		return err
+	}
+	p.assigned = true
+	p.value = val
+	return nil
+}
+
+func (p *Param) AsFloat32() float32 {
+	n, _ := strconv.ParseFloat(p.value, 32)
+	return float32(n)
+}
+
+func (p *Param) AsFloat64() float64 {
+	n, _ := strconv.ParseFloat(p.value, 64)
+	return n
+}
+
+func (p *Param) AsInt() int {
+	n, _ := strconv.ParseInt(p.value, 10, strconv.IntSize)
+	return int(n)
+}
+
+func (p *Param) AsInt8() int8 {
+	n, _ := strconv.ParseInt(p.value, 10, 8)
+	return int8(n)
+}
+
+func (p *Param) AsInt16() int16 {
+	n, _ := strconv.ParseInt(p.value, 10, 16)
+	return int16(n)
+}
+
+func (p *Param) AsInt32() int32 {
+	n, _ := strconv.ParseInt(p.value, 10, 32)
+	return int32(n)
+}
+
+func (p *Param) AsInt64() int64 {
+	n, _ := strconv.ParseInt(p.value, 10, 64)
+	return int64(n)
+}
+
+func (p *Param) AsUint() uint {
+	n, _ := strconv.ParseUint(p.value, 10, strconv.IntSize)
+	return uint(n)
+}
+
+func (p *Param) AsUint8() uint8 {
+	n, _ := strconv.ParseUint(p.value, 10, 8)
+	return uint8(n)
+}
+
+func (p *Param) AsUint16() uint16 {
+	n, _ := strconv.ParseUint(p.value, 10, 16)
+	return uint16(n)
+}
+
+func (p *Param) AsUint32() uint32 {
+	n, _ := strconv.ParseUint(p.value, 10, 32)
+	return uint32(n)
+}
+
+func (p *Param) AsUint64() uint64 {
+	n, _ := strconv.ParseUint(p.value, 10, 64)
+	return uint64(n)
+}
+
+func (p *Param) AsString() string {
+	return p.value
+}
+
+func (p *Param) AsStringSlice() []string {
+	return strings.Split(p.value, ",")
```

Comment:
JFYI: This change broke the local-gadget. The `-r // --runtimes` parameter has to be passed otherwise it doesn't work.

I have some ideas to fix it, will post them later on the day.